### ISSUE TITLE
clean updated data

### ISSUE THREAD

**Issue by lizzieinvancouver** [lizzieinvancouver/ospree#304]
It's time to start cleaning the updated OSPREE data. Thanks to everyone who helped enter the new data. I have merged it and added it here:

`MASTER branch: analyses/cleaning/cleanmerge_all.R`

I am going to focus asking for help (for now) from those helped originally with cleaning just to keep things moving. So I am wondering if:

@cchambe12 could you review # 4 and # 5?

@dbuona  Dan, could you update #3 (also, your code added a warning at some point and we should address this as we do other warnings, that is mention why it happens and if it is okay in the stop error message ...) and could you possible do #7 (including the part that is commented out ... so two source files)?

@AileneKane could you take a quick look at # 9?

If you have issues or concerns, let me know. Otherwise, please plan to finish by next Wednesday if possible.

**Comment by cchambe12** [comment]
@lizzieinvancouver Sure thing! Do you mean #5 and #6? So that would be clean_photo.R and clean_forcetemp.R

**Comment by lizzieinvancouver** [comment]
@cchambe12 (and @AileneKane @dbuona since I didn't make this very clear) Yes! The numbers are the numbers in cleanmerge_all.R, so yes, for you, Cat it would be clean_photo.R and clean_forcetemp.R. Thank you again! I am excited to get this moving.

**Comment by cchambe12** [comment]
@lizzieinvancouver @AileneKane @dbuona 

We also need to clean chilling/cleaning_chilltemp.R before moving forward.

**Comment by cchambe12** [comment]
clean_photo.R and clean_forecetemp.R should be all cleaned!

**Comment by AileneKane** [comment]
I just want to be sure tha ramos17 the Sciencia Horticultura paper from 2018 ("Chilling accumulation, dormancy release temperature, and the role of leaves in olive reproductive budburst: Evaluation using shoot explants"). Please let me know if I'm wrong about that!

**Comment by AileneKane** [comment]
clean_latlong.R is cleaned.

**Comment by lizzieinvancouver** [comment]
@AileneKane Yes! Note that all the new papers are in the same google drive folder as the other papers, they are just in their own folder (Papers_2016_2018 https://drive.google.com/open?id=1TsB7RN0Q2G8-91pDOfMgc-fo8wloj-Sm  ... let me know if you cannot see them).

**Comment by AileneKane** [comment]
@lizzieinvancouver I can see them- I just can't add to the folder, which is why I added the ramos17 supp to the main Ospree Papers folder (you may want to move it into the Papers_2016_2018 folder)

**Comment by lizzieinvancouver** [comment]
@AileneKane Sorry, I think you should have access now. Either way, can you please delete the Ramos17 paper from the main folder? Thanks!

**Comment by AileneKane** [comment]
@lizzieinvancouver done!

**Comment by lizzieinvancouver** [comment]
@cchambe12 for flynn18 I changed "daystoflowerbudburst" --> daystoflower (for respvar.simple). Please say if NOT okay.

@DarwinSodhi for man17, I changed 'thermaltimetohours' to 'cumchillinghours.' We usually use thermaltime to represent accumulated FORCING, but this is accumulated chilling, right?

@DeirdreLoughnan I made the following changes to papers you worked on for respvar.simple
ramos 17 -- "percentbudburst(vegetative +reproductive)" <- percentbudburst
ramos 17 -- "percentbudburst(reproductive)" <-- "percentflower"
fu19 -- "meanleafout" --> daystobudburst

Note the options are:

```
> unique(d$respvar.simple)
 [1] "growth"          "othernums"       "daystobudburst"  "thermaltime"    
 [5] "percentbudburst" "otherpercents"   "flowernumber"    "phenstage"      
 [9] "daystoflower"    "percentflower"   "notsureabout"    "fruitmass"  
```
So we're not looking to be exact, we're looking to group into the most appropriate category (you can check clean_respvar.R if it helps to see what is in each category).

Also, @DeirdreLoughnan can you please let me know how fu19 define leafout and please add the supp of any papers you used to the drive folder (e.g., you used the supp of fu19 so add that to the folder).

Thanks everyone!

**Comment by lizzieinvancouver** [comment]
@cchambe12 clean_photo.R removes 78 rows, is that fine?

**Comment by lizzieinvancouver** [comment]
@dbuona I took care of #7 in cleanmerge_all.R. Could you please do #3 in that file and #8 (same file) as soon as you have time? Thanks!

**Comment by DarwinSodhi** [comment]
@lizzieinvancouver yes you are correct it should be for chilling rather than forcing.

**Comment by lizzieinvancouver** [comment]
@DarwinSodhi Great! Thanks.

**Comment by cchambe12** [comment]
@lizzieinvancouver 
1. I'm okay with the respvar change for flynn18
2. The removed rows from clean_photo.R is fine! That's from the original ospree clean.

**Comment by lizzieinvancouver** [comment]
Goal to have new species list well before retreat and have these data super clean (including to budburst date).

**Comment by lizzieinvancouver** [comment]
Working now on bb_cleanmergeall.R ...

@DeirdreLoughnan vitra17 has thermaltimetobudburst ... which we generally try to convert to daystobudburst. Is there a calculation from the paper I could use to do this?

@DeirdreLoughnan @dbuona @cchambe12  the below studies appear to be ambient forcing ... is that correct? And can you confirma that if we want to estimate forcing we'd need daily climate data?
`c("fu18", "fu19", "prevey18", "richardson18")`

@cchambe12 If easy can you update this file: bb_analysis/cleaning/clean_checkdateofclimatedata.R
@cchambe12 Can you also let me know how much work updating this will be? source("bb_analysis/cleaning/clean_ambientforcingfromdailyclimate.R")

@dbuona Do we have any species complex code for the more stringent cut-off we once had (something like is in two studies and manipulates at least 2 cues or is in one study that manipulates three cues)? 

**To do still .**.. check bb_analysis/cleaning/clean_bbperctodays.R and make sure it still works well (deletes a lot of data just now).

**Comment by lizzieinvancouver** [comment]
@cchambe12 @dbuona Do either of you recall (or can find) our all notes on a more stringent cut-off? I recall something like two papers that manipulate two cues OR one paper that manipulates three cues and I could have sworn we wrote this down, but cannot find it. (I only have found [this](https://github.com/lizzieinvancouver/ospree/issues/232).)

**Comment by cchambe12** [comment]
@lizzieinvancouver There are some notes at the top of the bb_analysis/source/speciescomplex.R

## New Species complex f(x) that includes more studies
# criteria is just needs to be in one study now

##### Species were selected if they were in more than one study and if studies manipulated at least one cue
### Complexes were selected if (1) they were not already species (2) the genus was used in more than one study and (3) those studies used at least one cue

**Comment by DeirdreLoughnan** [comment]
@lizzieinvancouver 

vitra17 does not include an equation and it is not clear to me how else it would have been calculated.

fu19: yes, the forcing treatments are ambient+treatment temperatures, using outdoor climate-controlled chambers.

**Comment by lizzieinvancouver** [comment]
@cchambe12 Great! Thank you -- this is super helpful and just what I needed (@dbuona my above question is all set).

**Comment by cchambe12** [comment]
@lizzieinvancouver 

1. richardson18 is ambient or ambient + x
2.  bb_analysis/cleaning/clean_checkdateofclimatedata.R and bb_analysis/cleaning/clean_ambientforcingfromdailyclimate.R both look easy enough to fix
3. anzanello18 has different chill temperatures for day and night, I'm not sure how to address this.. with forcetemp  we have the two columns for day and night. For this study, there isn't any photoperiod the chilling is all in the dark.  Could we just take the average? One example of a treatment is 12 hours at 3 degC and 12 hours at 15 degC.

**Comment by lizzieinvancouver** [comment]
moving speciescomplex stuff to issue #308

**Comment by cchambe12** [comment]
I have cleaned all of the above! For now, I can't fix vitra17 photoperiod but if we are able to convert from thermaltimetodays to daystobudburst then I should be able to update it quickly.

**Comment by dbuona** [comment]
I checked clean_rampedandexpforcing.R. It seemed to me that the only changes from addition data were the addition of 4 ambient forcing conditions and if I am not mistaken, those will get cleaned in clean_ambientforcingfromdailyclimate.R. @cchambe12 is that correct?

**Comment by cchambe12** [comment]
That's correct!

**Comment by lizzieinvancouver** [comment]
@DeirdreLoughnan Still waiting on info on prevey18 (see above), Mika entered this but can you confirm and @dbuona same for you for fu18: it was ambient forcing ... is that correct? And can you confirma that if we want to estimate forcing we'd need daily climate data?

**Comment by lizzieinvancouver** [comment]
Re: clean_rampedandexpforcing.R ... @cchambe12 @dbuona Thanks! Cat, are those papers cleaned currently in clean_ambientforcingfromdailyclimate.R?

**Comment by lizzieinvancouver** [comment]
Checking on the above -- we need to get this stuff done soon ... @cchambe12 Let us know if you need help!
@cchambe12 check out clean_checkdateofclimatedata.R, clean_ambientforcingfromdailyclimate.R, clean_checkdateofclimatedata.R, clean_photoperiod.R (re-assign to others as needed ...)
@MoralesCastilla Check clean_bbperctodays.R again, why losing SO many rows now (with new studies)? Okay?

**Comment by lizzieinvancouver** [comment]
Only paper we need to add to clean_thermaltimetodays.R is vitra17 ... waiting on info -- see issue #309 ... also I need to update notes in bbcleaning_README.txt once I have back on vitra17, prevey and fu18.

**Comment by cchambe12** [comment]
@lizzieinvancouver sorry everyone these have been done for a while now I just forgot to comment here!

**Comment by lizzieinvancouver** [comment]
@lizzieinvancouver Will check prevey18 and fu18 ... @MoralesCastilla will send Lizzie the rows lost in clean_bbperctodays.R

**Comment by lizzieinvancouver** [comment]
Lizzie reviewed fu18 -- it looks like it is entered correctly (ambient+ experiments).

Here's the remaining issues to check on this issue (then we can close it, note that vitra thermal time and bbpercdays are handled in other issues so don't need to review here):

@cchambe12 check out clean_checkdateofclimatedata.R, clean_ambientforcingfromdailyclimate.R, clean_checkdateofclimatedata.R, clean_photoperiod.R (re-assign to others as needed ...)

This issue can be closed by Cat if confirmed the above are all dealt with. Thanks!
